### PR TITLE
[Semana 8] Testes

### PR DESCRIPTION
**Pull request by NickMendes**

### CODE REVIEW

File: account/src/test/routes/UserRoute.test.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            })
+            .expect(201);
+        idResponse = response.body['_id'];    
+    });
+
+    it('Should return error if body is empty', async () => {
+        await request(app)
+            .post('/admin/user')
+            .send({})
+            .expect(500);
```
Comment (DPechetti) [inline account/src/test/routes/UserRoute.test.js:53]: is this really supposed to be 500?

File: account/src/test/routes/UserRoute.test.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        expect(response.body.name).toEqual('Testando Test');
+        expect(response.body.password).toEqual('testando123');
+    });
+});
+
+describe('DELETE route /admin/user/:id', () => {
+    it('Should delete the selected user', async () => {
+        await request(app)
+            .delete(`/admin/user/${idResponse}`)
+            .expect(202);
```
Comment (DPechetti) [inline account/src/test/routes/UserRoute.test.js:87]: is this really supposed to be 202?

File: finance-mongo/src/test/routes/CupomRoute.test.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+});
+
+describe('PUT route /admin/cupom/:id', () => {
+    it('Should update information on the selected cupom', async () => {
+        await request(app)
+            .patch(`/admin/cupom/${idResponse}`)
+            .send({
+                name: 'Testando'
+            })
+            .expect(202);
```
Comment (DPechetti) [inline finance-mongo/src/test/routes/CupomRoute.test.js:62]: is this really supposed to be 202?

File: finance-mongo/src/test/routes/CupomRoute.test.js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            .expect(200);
+        expect(response.body.user_info.name).toEqual('Testando');
+    });
+});
+
+describe('DELETE route /admin/cupom/:id', () => {
+    it('Should delete the selected cupom', async () => {
+        await request(app)
+            .delete(`/admin/cupom/${idResponse}`)
+            .expect(202);
```
Comment (DPechetti) [inline finance-mongo/src/test/routes/CupomRoute.test.js:74]: is this really supposed to be 202?

File: finance-mongo/src/test/routes/PaymentRoute.test.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            })
+            .expect(201);
+        idResponse = response.body['_id'];
+    });
+
+    it('Should return error if body is empty', async () => {
+        await request(app)
+            .post('/admin/payment')
+            .send({})
+            .expect(500);
```
Comment (DPechetti) [inline finance-mongo/src/test/routes/PaymentRoute.test.js:54]: is this really supposed to be 500?

File: finance-mongo/src/test/routes/PaymentRoute.test.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+});
+
+describe('PUT route /admin/payment/:id', () => {
+    it('Should update information on the selected payment', async () => {
+        await request(app)
+            .patch(`/admin/payment/${idResponse}`)
+            .send({
+                name: 'Testando'
+            })
+            .expect(202);
```
Comment (DPechetti) [inline finance-mongo/src/test/routes/PaymentRoute.test.js:82]: is this really supposed to be 202?

File: finance-mongo/src/test/routes/PaymentRoute.test.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+            .expect(200);
+        expect(response.body.name).toEqual('Testando');
+    });
+});
+
+describe('DELETE route /admin/payment/:id', () => {
+    it('Should delete the selected payment', async () => {
+        await request(app)
+            .delete(`/admin/payment/${idResponse}`)
+            .expect(202);
```
Comment (DPechetti) [inline finance-mongo/src/test/routes/PaymentRoute.test.js:94]: is this really supposed to be 202?

File: account/src/test/routes/UserRoute.test.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+describe('PUT route /admin/user/:id', () => {
+    it('Should update information on the selected user', async () => {
+        await request(app)
+            .put(`/admin/user/${idResponse}`)
+            .send({
+                name: 'Testando Test',
+                password: 'testando123'
+            })
+            .expect(202);
```
Comment (DPechetti) [inline account/src/test/routes/UserRoute.test.js:74]: is this really supposed to be 202?

File: finance/api/tests/PaymentRoute.test.js [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+                cvv: '598'
+            })
+            .expect(201);
+        idResponse = response.body.id;
+    });
+    it('Should return error if body is empty', async () => {
+        await request(api)
+            .post('/payments')
+            .send({})
+            .expect(500);
```
Comment (DPechetti) [inline finance/api/tests/PaymentRoute.test.js:34]: is this really supposed to be 500?

File: finance/api/tests/PaymentRoute.test.js [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+            .expect(200);
+        expect(response.body.status).toEqual('CONFIRMADO');
+    });
+});
+
+describe('DELETE route /payments/:id', () => {
+    it('Should delete the selected payment', async () => {
+        await request(api)
+            .delete(`/payments/${idResponse}`)
+            .expect(202);
```
Comment (DPechetti) [inline finance/api/tests/PaymentRoute.test.js:68]: is this really supposed to be 202?

File: order/src/test/routes/SalesRoute.test.js [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+});
+
+describe('PUT route /admin/sale/:id', () => {
+    it('Should update information on the selected sale', async () => {
+        await request(app)
+            .put(`/admin/sale/${idResponse}`)
+            .send({
+                cep: '123569-587'
+            })
+            .expect(202);
```
Comment (DPechetti) [inline order/src/test/routes/SalesRoute.test.js:71]: is this really supposed to be 202?

File: order/src/test/routes/SalesRoute.test.js [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+            .expect(200);
+        expect(response.body.delivery_address.cep).toEqual('123569-587');
+    });
+});
+
+describe('DELETE route /admin/sale/:id', () => {
+    it('Should delete the selected sale', async () => {
+        await request(app)
+            .delete(`/admin/sale/${idResponse}`)
+            .expect(202);
```
Comment (DPechetti) [inline order/src/test/routes/SalesRoute.test.js:83]: is this really supposed to be 202?

File: package.json [hunk 13]
```diff
@@ -0,0 +1,13 @@
+{
+  "dependencies": {
+    "jest": "^29.4.3"
```
Comment (DPechetti) [inline package.json:3]: do you need the jest here?

File: product/src/tests/routes/categoryRoute.test.js [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+});
+
+describe('PUT route /admin/categories/:id', () => {
+    it('Should update information on the selected category', async () => {
+        await request(app)
+            .put(`/admin/categories/${idResponse}`)
+            .send({
+                name: 'Testando'
+            })
+            .expect(202);
```
Comment (DPechetti) [inline product/src/tests/routes/categoryRoute.test.js:61]: is this really supposed to be 202?

File: product/src/tests/routes/categoryRoute.test.js [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+        const response = await request(app)
+            .get(`/categories/${idResponse}`)
+            .expect(200);
+        expect(response.body.name).toEqual('Testando');
+    });
+    it('Should return error if body is empty', async () => {
+        await request(app)
+            .put(`/admin/categories/${idResponse}`)
+            .send({})
+            .expect(500);
```
Comment (DPechetti) [inline product/src/tests/routes/categoryRoute.test.js:71]: is this really supposed to be 500?

File: product/src/tests/routes/categoryRoute.test.js [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+});
+
+describe('PATCH route /admin/categories/:id', () => {
+    it('Should update status on the selected categories', async () => {
+        await request(app)
+            .patch(`/admin/categories/${idResponse}`)
+            .send({
+                status: 'Ativa'
+            })
+            .expect(202);
```
Comment (DPechetti) [inline product/src/tests/routes/categoryRoute.test.js:82]: is this really supposed to be 202?

File: product/src/tests/routes/categoryRoute.test.js [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+            .expect(400);
+    });
+});
+
+
+describe('DELETE route /admin/categories/:id', () => {
+    it('Should delete the selected categories', async () => {
+        await request(app)
+            .delete(`/admin/categories/${idResponse}`)
+            .expect(202);
```
Comment (DPechetti) [inline product/src/tests/routes/categoryRoute.test.js:101]: is this really supposed to be 202?

File: product/src/tests/routes/productRoute.test.js [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+
+describe('PUT route /admin/product/:id', () => {
+    it('Should update information on the selected product', async () => {
+        await request(app)
+            .put(`/admin/product/${idResponse}`)
+            .send({
+                name: 'Play Station 6',
+                unit_price: 10000.00
+            })
+            .expect(202);
```
Comment (DPechetti) [inline product/src/tests/routes/productRoute.test.js:66]: is this really supposed to be 202?

File: product/src/tests/routes/productRoute.test.js [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+            .expect(200);
+        expect(response.body.name).toEqual('Play Station 6');
+    });
+});
+
+describe('DELETE route /admin/product/:id', () => {
+    it('Should delete the selected product', async () => {
+        await request(app)
+            .delete(`/admin/product/${idResponse}`)
+            .expect(202);
```
Comment (DPechetti) [inline product/src/tests/routes/productRoute.test.js:78]: is this really supposed to be 202?

**Review by DPechetti — CHANGES_REQUESTED** [review]

File: account/package.json [hunk 20]
```diff
@@ -5,9 +5,9 @@
   "main": "server.js",
   "type": "module",
   "scripts": {
-    "dev": "nodemon server.js",
     "start": "node server.js",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "node --experimental-vm-modules node_modules/jest/bin/jest.js --verbose",
```
Comment (tiagolpadua) [inline account/package.json:9]: Seria interessante ter um script específico para o eslint, para facilitar sua execução, algo como:
"lint": "eslint src"

File: account/src/controllers/UserController.js [hunk 21]
```diff
@@ -1,63 +1,63 @@
...
-        res.status(202).send({ message: 'User deleted success' });
-      } else {
-        res.status(500).send({ message: err.message });
-      }
-    });
-  };
+    static getAllUsers = (_req, res) => {
+        users.find((err, user) => {
+            if(err) {
+                res.status(400).send({ message: err.message });
```
Comment (tiagolpadua) [inline account/src/controllers/UserController.js:7]: Erros na faixa do 400 em geral indicam algum erro nos dados enviados na requisição, neste caso, seria mais adequado um erro na faixa do 500

File: account/src/controllers/UserController.js [hunk 22]
```diff
@@ -1,63 +1,63 @@
...
+            } else {
+                res.status(200).json(user);
```
Comment (tiagolpadua) [inline account/src/controllers/UserController.js:9]: Seria bom tratar o caso de entidade não encontrada retornando 404

File: account/src/controllers/UserController.js [hunk 23]
```diff
@@ -1,63 +1,63 @@
...
+            if(err) {
+                res.status(400).send({ message: err.message });
+            } else {
+                res.status(200).send(user);
+            }
+        });
+    };
+
+    static createUser = (req, res) => {
+        let user = new users(req.body);
```
Comment (tiagolpadua) [inline account/src/controllers/UserController.js:27]: Prefira utilizar nomes de classes e modelos no singular e com a primeira letra em maiúsculo, como no exemplo do mongoose: https://mongoosejs.com/docs/models.html#compiling

**Comment by tiagolpadua** [comment]
Card1 - Testando os endpoints da API de produtos	Ok
Card2 - Testando os endpoints da API de pagamentos	Ok
Card3 - Configuração do ES Lint	Ok
Card4 - Testando os endpoints da API de usuários	Ok
Card5 - Testando os endpoints da API de pedidos	Ok

**Pull request merged**